### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by varvarenique**
https://trello.com/c/bE8UVx1d/13-pack-the-service-as-a-docker-container
Here is link for my image: https://hub.docker.com/r/varvarenique/delivery_sys
Update: https://hub.docker.com/r/varvarenique/delivery

What I use for run: docker run -it -p 8888:5000 --name container_delivery -v "D:/Exadel/restaurants.json:/etc/data/restaurants.json" -e PATH_FOR_INITIAL_DATA="/etc/data/restaurants.json" varvarenique/delivery

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -129,5 +129,5 @@ def create_app():
 
 
 if __name__ == '__main__':
-    app.run()
+    app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
```
Comment (minotru) [inline app.py]: should be a part of `Config`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM python:3
+
+ARG PATH_FOR_INITIAL_DATA="restaurants.json"
```
Comment (minotru) [inline Dockerfile]: `ARG` is used to specify Docker build-time argument, while the path to the data should be set in runtime with `ENV`. 

in general, `restaurants.json` should not even be added to the `Docker` image. instead, `restaurants.json` should be attached as a volume and a proper environment variable should be set during `docker run`.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+COPY . /app
+
+# install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# tell the port number the container should expose
+EXPOSE 5000
+
+# run the command
+CMD ["python", "./app.py"]
```
Comment (minotru) [inline Dockerfile]: `CMD` is used to specify a default command to execute when contain is running. this is fine for  some cases (e.g. for `linux` image to run `bash` terminal by default), but in general `ENTRYPOINT` is preferred:
https://goinbigdata.com/docker-run-vs-cmd-vs-entrypoint/
"The bottom line"

**Comment by minotru** [comment]
> Here is link for my container

Actually, it is `image`, not `container`: Docker builds `image` that becomes `container` when it is run :)

File: app.py (hunk 1 above)
Comment (varvarenique) [reply app.py]: fixed

File: Dockerfile (hunk 3 above)
Comment (varvarenique) [reply Dockerfile]: fixed

File: Dockerfile (hunk 2 above)
Comment (varvarenique) [reply Dockerfile]: fixed

**Review by minotru — APPROVED** [review]

**Pull request merged**